### ISSUE TITLE
Rule proposal: `consistent-non-exists-index-check`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1024]
Similar to [`explicit-length-check`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/explicit-length-check.md), enforce consistent style when check `index === -1` or `index !== -1`

## Fail

```js
const index = foo.indexOf('bar');

if (index < 0) {}
```

```js
const index = foo.indexOf('bar');

if (index >= 0) {}
```

## Pass

```js

const index = foo.indexOf('bar');

if (index === -1) {}
```

```js

const index = foo.indexOf('bar');

if (index !== -1) {}
```

But this is hard to detect, since we already [enforce `includes`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/prefer-includes.md) if `indexOf()` is in a position as `boolean`.

**Comment by sindresorhus** [comment]
`Array#includes` would be better for this, which we already have a rule for, so I'm not sure the point of this rule?

**Comment by fisker** [comment]
Sometime, we want index, but need check if it's `-1` before using it, example https://github.com/sindresorhus/eslint-plugin-unicorn/blob/5ba0f8379d06e581c614245200bdbd3d935822a7/rules/no-array-push-push.js#L42-L44

**Comment by sindresorhus** [comment]
Ok. This rule makes sense then. We should make what you said above clear in the rule docs though.

**Comment by sindresorhus** [comment]
This is now accepted.

**Comment by axetroy** [comment]
Maybe `if (foo.indexOf('bar') < 0) {}` should also be checked?

**Comment by sindresorhus** [comment]
> Maybe, should if (foo.indexOf('bar') < 0) {} also be checked?

No point, since it's already caught by `prefer-includes`.
